crypto/src/e2e_identity/identity.rs

```diff
@@ -421,6 +529,119 @@ mod tests {
+                    alice_android_central
+                        .mls_central
+                        .new_conversation(&id, case.credential_type, case.cfg.clone())
+                        .await
+                        .unwrap();
+                    alice_android_central
+                        .mls_central
+                        .invite_all(
+                            &case,
+                            &id,
+                            [&mut alice_ios_central.mls_central, &mut bob_android_central.mls_central],
+                        )
+                        .await
+                        .unwrap();
+
+                    let nb_members = alice_android_central
+                        .mls_central
+                        .get_conversation_unchecked(&id)
+                        .await
+                        .members()
+                        .len();
+                    assert_eq!(nb_members, 3);
+
+                    assert_eq!(
+                        alice_android_central.mls_central.get_user_id(),
+                        alice_ios_central.mls_central.get_user_id()
+                    );
+
+                    // Finds both Alice's devices
+                    let alice_user_id = alice_android_central.mls_central.get_user_id();
+                    let alice_identities = alice_android_central
+                        .mls_central
+                        .get_user_identities(&id, &[alice_user_id.clone()])
+                        .await
+                        .unwrap();
+                    assert_eq!(alice_identities.len(), 1);
+                    let identities = alice_identities.get(&alice_user_id).unwrap();
+                    assert_eq!(identities.len(), 2);
+
+                    // Finds Bob only device
+                    let bob_user_id = bob_android_central.mls_central.get_user_id();
+                    let bob_identities = alice_android_central
+                        .mls_central
+                        .get_user_identities(&id, &[bob_user_id.clone()])
+                        .await
+                        .unwrap();
+                    assert_eq!(bob_identities.len(), 1);
+                    let identities = bob_identities.get(&bob_user_id).unwrap();
+                    assert_eq!(identities.len(), 1);
+
+                    // Finds all devices
+                    let all_identities = alice_android_central
+                        .mls_central
+                        .get_user_identities(&id, &[alice_user_id.clone(), bob_user_id.clone()])
+                        .await
+                        .unwrap();
+                    assert_eq!(all_identities.len(), 2);
+                    let alice_identities = alice_identities.get(&alice_user_id).unwrap();
+                    assert_eq!(alice_identities.len(), 2);
+                    let bob_identities = bob_identities.get(&bob_user_id).unwrap();
+                    assert_eq!(bob_identities.len(), 1);
```

Comment:
This looks weird. And I realize this is just copied from the original test, but I think we should nevertheless address it.

Here we just check that the length of `all_identities` is 2.
But then, we repeat the same checks for Bob and Alice that we had above. So I'm not really sure what this is supposed to test. Maybe the original author intended to write something like the following?

```rust
let all_identities = ...
assert_eq!(all_identities.get(&alice_user_id).unwrap().len(), 2);
assert_eq!(all_identities.get(&bob_user_id).unwrap().len(), 1);
```
That is, to use `all_identities` for the checks.